### PR TITLE
crowdstrike_oauth_api_dynamic_analysis_playbooks

### PR DESCRIPTION
**Pull request by tccontre**
##### Please check all that apply:
- [x] New playbook
- [ ] Bugfix
- [ ] Feature add
- [ ] Code style update (formatting, renaming)
- [ ] Documentation
- [ ] Other (please describe): 

# Release Notes
Replace the following list with release notes that describe the high level components of the PR:
- [x]  playbooks/CrowdStrike_OAuth_API_Dynamic_Analysis.json
- [x]  new file:   playbooks/CrowdStrike_OAuth_API_Dynamic_Analysis.png
- [x]  new file:   playbooks/CrowdStrike_OAuth_API_Dynamic_Analysis.py
- [x]  new file:   playbooks/CrowdStrike_OAuth_API_Dynamic_Analysis.yml

# Playbook quality checklist
Please check if your PR fulfills the following requirements.

### Requirements for Settings

- [x] Playbook name is A-Z in Title case with underscores between words. (e.g. MS_Graph_Search_and_Purge)
- [ ] Category in Title case with spaces between words (e.g. Identifier Reputation Analysis)
- [x] Description is free of grammatical errors and describe what the playbook does.
- [x] Notes list any setup required on the third-party API as well as intended areas for customization.
- [ ] Label is set to '\*'
    
### Requirements for all playbooks

- [ ] Playbook block count not greater than 15 (not including Start and End blocks).
- [ ] No more than 3 branching paths.
- [x] If referencing a custom list, Notes document what the expected values are in that custom list.

### Requirements for all playbook blocks

- [x] All blocks have a custom name no more than 4 words, all lowercase, and separated by space (e.g. close workbook task)
- [x] All blocks that support a Notes Tooltip have it filled out. Must be grammatically correct and describes the intended purpose of that block.
- [x] Where custom code is used, block notes indicate presence of custom code (e.g. "This block uses custom code")
- [x] No block is disabled by custom code
- [x] Custom code is documented with notes

##### Action

- [x] Use apps available on Splunkbase
- [x] Use asset names that are the app name, all lowercase separated by underscores (e.g. Azure AD Graph becomes azure_ad_graph)

##### Utility 

- [x] Block is using community version

##### Playbook

- [x] Block is using local version

##### Input playbooks
- [x] Start blocks use ocsf variable names and a minimum of one data type per variable name (e.g. device (type: host name))
- [x] Has at least one category tag (e.g. reputation)
- [x] Playbook has a tag for each vendor app used

# Other considerations (PR type specific)
- [x] If new playbook, there is a screenshot ending in .png with the same name as the playbook .json
- [x] Playbook major minor version matches repo (e.g. 5.5 != 6.0)
- [x] PR contains both .py and .json

---
Thanks for contributing!

### CODE REVIEW

File: playbooks/CrowdStrike_OAuth_API_Dynamic_Analysis.py [hunk 1]
```diff
@@ -0,0 +1,1410 @@
...
+    ################################################################################
+    ## Custom Code End
+    ################################################################################
+
+    phantom.save_run_data(key="build_android_file_output:observable_array", value=json.dumps(build_android_file_output__observable_array))
+
+    return
+
+
+@phantom.playbook_block()
```
Comment (tccontre) [inline playbooks/CrowdStrike_OAuth_API_Dynamic_Analysis.py:1158]: feel free to add the actual mac sandbox for mac file detonation part.

**Review by P4T12ICK — CHANGES_REQUESTED** [review]
- [x] Add Category Dynamic Analysis
- [x] Vault ID Observables vault_id: _vault_id need to be renamed to value: _vault_id.
- [x] Rename tag reputation to sandbox and update it in observables.
- [x] In Playbook Settings, tag for vendor app need to be added.
- [x] End Block need to be configured to return observables and report

**Comment by tccontre** [comment]
> * Add Category Dynamic Analysis
> * Vault ID Observables vault_id: _vault_id need to be renamed to name: _vault_id.
> * Do we need a new category tag instead of reputation? Maybe dynamic_analysis? Reputation is used for another set of playbooks.
> * In Playbook Settings, tag for vendor app need to be added.

**Comment by tccontre** [comment]
> * Add Category Dynamic Analysis
> * Vault ID Observables vault_id: _vault_id need to be renamed to value: _vault_id.
> * Rename tag reputation to sandbox and update it in observables.
> * In Playbook Settings, tag for vendor app need to be added.
> * End Block need to be configured to return observables and report

all done. :)

**Review by P4T12ICK — CHANGES_REQUESTED** [review]
- [x] Remove / Comment debug statements in code blocks

Testing was good. vault_id can only work with a list of same file types (e.g. windows). But I think there is no other way to handle it.

**Review by P4T12ICK — APPROVED** [review]

**Pull request merged**